### PR TITLE
feat: add shpotify spec

### PR DESCRIPTION
**Pull request by paolorossig**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Feature

**What is the current behavior? (You can also link to an open issue here)**
`spotify` commands are currently not-supported

**What is the new behavior (if this is a feature change)?**
 Autocompletion support for `spotify` commands

**Additional info:**
Link of [shpotify](https://github.com/hnarayanan/shpotify)

### CODE REVIEW

File: src/spotify.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
+const completionSpec: Fig.Spec = {
```
Comment (fedeci) [inline src/spotify.ts:1]:
```suggestion
// https://github.com/hnarayanan/shpotify
const completionSpec: Fig.Spec = {
```

File: src/spotify.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+          description: "Finds a playlist by name and plays it",
+          args: {
+            name: "playlist name",
+            description: "The playlist name you want to play",
+            isOptional: false,
+          },
+        },
+        {
+          name: "uri",
+          description: "Plau songs from specific uri",
```
Comment (fedeci) [inline src/spotify.ts:43]:
```suggestion
          description: "Play songs from specific uri",
```

File: src/spotify.ts [hunk 3]
```diff
@@ -0,0 +1,139 @@
+const completionSpec: Fig.Spec = {
+  name: "spotify",
+  description: "CLI to use Spotify from the terminal",
+  subcommands: [
+    {
+      name: "play",
+      description: "Resumes playback where Spotify last left off",
+      args: {
+        name: "song name",
+        description: "Finds a song by name and plays it",
```
Comment (fedeci) [inline src/spotify.ts:10]: The arg description is meant to refer to the arg content, not to the action performed by the subcommand/option of which the argument is part:
```suggestion
        description: "The name of the song to start playing",
```

File: src/spotify.ts [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        isOptional: true,
+      },
+      subcommands: [
+        {
+          name: "album",
+          description: "Finds an album by name and plays it",
+          args: {
+            name: "album name",
+            description: "The album name you want to play",
+            isOptional: false,
```
Comment (fedeci) [inline src/spotify.ts:20]: This is implicit for args
```suggestion
```

File: src/spotify.ts [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+          },
+        },
+        {
+          name: "artist",
+          description: "Finds an artist by name and plays it",
+          args: {
+            name: "artist name",
+            description: "The artist name you want to play",
+            isOptional: false,
```
Comment (fedeci) [inline src/spotify.ts:29]:
```suggestion
```

File: src/spotify.ts [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+          },
+        },
+        {
+          name: "list",
+          description: "Finds a playlist by name and plays it",
+          args: {
+            name: "playlist name",
+            description: "The playlist name you want to play",
+            isOptional: false,
```
Comment (fedeci) [inline src/spotify.ts:38]:
```suggestion
```

File: src/spotify.ts [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+          },
+        },
+        {
+          name: "uri",
+          description: "Plau songs from specific uri",
+          args: {
+            name: "uri",
+            description: "The uri of the song you want to play",
+            isOptional: false,
```
Comment (fedeci) [inline src/spotify.ts:47]:
```suggestion
```

File: src/spotify.ts [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+      name: "replay",
+      description: "Replays the current track from the beginning",
+    },
+    {
+      name: "pos",
+      description: "Jumps to a time (in secs) in the current song",
+      args: {
+        name: "time",
+        description: "Exact time in secs to jump in",
+        isOptional: false,
```
Comment (fedeci) [inline src/spotify.ts:70]:
```suggestion
```

File: src/spotify.ts [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+
+      args: {
+        name: "amount",
+        description: "Sets the volume to an amount between 0 and 100",
+        isOptional: true,
+      },
+      subcommands: [
+        { name: "up", description: "Increases the volume by 10%" },
+        { name: "down", description: "Decreases the volume by 10%" },
+      ],
```
Comment (fedeci) [inline src/spotify.ts:97]: This should be condensed to the following structure:
```suggestion
      args: {
        name: "amount",
        description: "Sets the volume to an amount between 0 and 100",
        isOptional: true,
        suggestions: [
          { name: "up", description: "Increases the volume by 10%" },
          { name: "down", description: "Decreases the volume by 10%" },
        ],
      },
```

**Review by fedeci — CHANGES_REQUESTED** [review]
Thanks for this spec! I am searching for a Spotify CLI and will try this one😄
 I left some comments to be resolved before merging this PR, would you mind also removing the third person from descriptions?
e.g.
"Resumes playback where Spotify last left off" should become "Resume playback where Spotify last left off"
This is super nitty but would improve consistency with other specs!

**Review by paolorossig — COMMENTED** [review]
Apply suggestions from code review

**Comment by paolorossig** [comment]
> Resume

You're welcome! You definitely have to try it.
Sorry, it was my first OSS PR and I forgot to remove the third person from descriptions. I'm on it.

**Comment by fedeci** [comment]
Don't worry man, you did a great job!

**Review by fedeci — APPROVED** [review]
Thanks!

**Pull request merged**